parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ColumnChunkPageWriteStore.java

```diff
@@ -65,10 +66,20 @@
     private Set<Encoding> encodings = new HashSet<Encoding>();
 
     private Statistics totalStatistics;
+    private final ByteBufferAllocator allocator;
 
-    private ColumnChunkPageWriter(ColumnDescriptor path, BytesCompressor compressor, int pageSize) {
+    // TODO - look back at this, the ByteBuffer code changes involved passing an allocator here
+    // the way this was refactored with the ConcatenatingByteArrayCollector, it is supposed
+    // to be avoiding copies
+    // I assume that this is actually going to be making a copy from off-heap ByteBuffers into
+    // byte arrays, I think the ConcatenatingByteArrayCollector likely should be refactored
+    // to be maintaining a list of ByteBuffers instead of byte arrays
```

Comment:
I took another look, the ConcatenatingByteArrayCollector (and BytesInput its superclass) only has a writeAllTo method that takes an OutputStream as a parameter. This interface only has a write method that takes a byte[]. Looking at the docs for the Hadoop FSDataOutputStream, it looks like even if we refactored the code everywhere to refer to it (or some kind of wrapper class to keep compatiblity with 1.x), there is no method that takes a bytebuffer for writing to the stream like the read API now has in Hadoop 2. I'll remove the TODO for now.